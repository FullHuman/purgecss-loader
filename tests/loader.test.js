const { join } = require('path')
const webpack = require('webpack')
const MemoryFileSystem = require('memory-fs')

test('loader removes unused CSS rules', done => {
  const compiler = webpack({
    entry: join(__dirname, './fixtures/example.css'),
    module: {
      rules: [
        {
          test: /\.css$/,
          use: [
            {
              loader: '@ianwalter/callback-loader',
              options: {
                callback: function (content) {
                  expect(content).toMatchSnapshot()
                  done()
                }
              }
            },
            {
              loader: join(__dirname, '../'),
              options: { content: [join(__dirname, 'fixtures/example.html')] }
            }
          ]
        }
      ]
    }
  })
  compiler.outputFileSystem = new MemoryFileSystem()
  compiler.run()
})
