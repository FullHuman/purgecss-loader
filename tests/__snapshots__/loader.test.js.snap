// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader removes unused CSS rules 1`] = `
"h2.example {
  font-size: 2rem;
}
h5.example {
  font-size: 1.25rem;
}
"
`;
